src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java

```diff
@@ -18,17 +23,65 @@ public void onPlayerMove(PlayerMoveEvent event) {
-            if (event.getPlayer().getLocation().getY() < LocationUtils.getMinHeight(event.getPlayer().getWorld())) {
-                VoidEnhancementData voidEnhancementData = IridiumSkyblock.getInstance().getEnhancements().voidEnhancement.levels.get(IridiumSkyblock.getInstance().getTeamManager().getTeamEnhancement(island, "void").getLevel());
-                if (voidEnhancementData != null && voidEnhancementData.enabled) {
-                    if (IridiumSkyblock.getInstance().getTeamManager().teleport(event.getPlayer(), island.getHome(), island)) {
-                        event.getPlayer().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().voidTeleport
-                                .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)
-                        ));
+            if (!(event.getPlayer().getLocation().getY() < LocationUtils.getMinHeight(event.getPlayer().getWorld()))) {
+                return;
+            }
+
+            VoidEnhancementData voidEnhancementData = IridiumSkyblock.getInstance()
+                    .getEnhancements().voidEnhancement.levels
+                    .get(IridiumSkyblock.getInstance().getTeamManager().getTeamEnhancement(island, "void").getLevel());
+
+            if (voidEnhancementData == null || !voidEnhancementData.enabled) {
+                return;
+            }
+
+            if (!IridiumSkyblock.getInstance().getTeamManager().teleport(event.getPlayer(), island.getHome(), island)) {
+                return;
+            }
+
+            event.getPlayer().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().voidTeleport
+                    .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+
+            HashMap<ItemStack, Integer> lostItemsMap = new HashMap<>();
+            if (voidEnhancementData.itemLossChance > 0) {
+                for (ItemStack item : event.getPlayer().getInventory().getContents()) {
+                    if (item == null) {
+                        continue;
                     }
+
+                    ItemStack originalItem = item.clone();
+
+                    for (int i = 0; i < item.getAmount(); i++) {
+                        if (Math.random() * 100 > voidEnhancementData.itemLossChance) {
+                            continue;
+                        }
+
+                        lostItemsMap.compute(originalItem, (key, value) -> value == null ? 1 : value + 1);
+                    }
+
+                    item.setAmount(item.getAmount() - lostItemsMap.getOrDefault(originalItem, 0));
                 }
+
+                if (lostItemsMap.isEmpty()) {
+                    return;
+                }
+
+                IridiumSkyblock.getInstance().getDatabaseManager().getLostItemsTableManager().addEntry(new LostItems(
+                        event.getPlayer().getUniqueId(), lostItemsMap.keySet().toArray(new ItemStack[0])));
```

Comment:
Should store this as an array, also the amount of this Item should reflect the amount lost, not the origional amount